### DISCUSSION TITLE
HUB12 Mode ?

### DISCUSSION THREAD

**Discussion by diplfranzhoepfinger** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#526] [General]
Hi, 

i got this Panel as well: 
https://www.reichelt.de/arduino-led-matrix-32-x-16-leds-rot-ard-led-32x16-rd-p282619.html?CCOUNTRY=445&LANGUAGE=de&nbc=1&&r=1

https://www.reichelt.de/arduino-led-matrix-32-x-16-leds-weiss-ard-led-32x16-wt-p282620.html?CCOUNTRY=445&LANGUAGE=de&nbc=1&&r=1

The have HUB12 Connection, 
with only 1 Color.

**Comment by board707** [comment]
Hi
 For monochrome panels you need a different library - for example DMD.h or DMD2.h

But the panel on the photo looks quite old (dip leds used) , perhaps you will experience troubles with it.

**Reply by diplfranzhoepfinger** [reply]
Where to find this ? 

i found one: https://github.com/pixelmatix/SmartMatrix

i have also one with SMD Chips.

**Reply by board707** [reply]
I am not sure that SmartMatrix will work with HUB12 panel, 

You can try DMD32 lib (I haven't test it myself)
https://github.com/Qudor-Engineer/DMD32

**Comment by rwahyur** [comment]
Hai @mrfaptastic 
@board707 
@beta-tester 


What about if we change some code in the virtual matrix.h to change the coordinate to use hub12 like smartmatrix library ?

Because the library from @mrfaptastic has supported 1/4 scan on panel that have 64, 32 or 16  pixel height


https://community.pixelmatix.com/t/monochrome-32-16-p10-hub12-led-matrix-control-with-esp32/812/9?page=2

**Reply by mrcodetastic** [reply]
If you want to submit a pull request that doesn't break support for everything else. Please do.

**Reply by Ammarfrfr** [reply]
Hey has he uploaded it?

**Comment by rwahyur** [comment]
Hai @mrfaptastic
@board707
@beta-tester




void MapSinglePanel(int x, int y, int *matrix_x, int *matrix_y) const {
	if (y / 4 == 0) {
		*matrix_y = y % 4;
		if (x / 8 == 0) {
			*matrix_x = x + 24;
		} else if (x / 8 == 1) {
			*matrix_x = x + 48;
		} else if (x / 8 == 2) {
			*matrix_x = x + 72;
		} else if (x / 8 == 3) {
			*matrix_x = x + 96;
		}
	} else if (y / 4 == 1) {
		*matrix_y = y % 4;
		if (x / 8 == 0) {
			*matrix_x = x + 16;
		} else if (x / 8 == 1) {
			*matrix_x = x + 40;
		} else if (x / 8 == 2) {
			*matrix_x = x + 64;
		} else if (x / 8 == 3) {
			*matrix_x = x + 88;
		}

	} else if (y / 4 == 2) {
		*matrix_y = y % 4;
		if (x / 8 == 0) {
			*matrix_x = x + 8;
		} else if (x / 8 == 1) {
			*matrix_x = x + 32;
		} else if (x / 8 == 2) {
			*matrix_x = x + 56;
		} else if (x / 8 == 3) {
			*matrix_x = x + 80;
		}

	} else if (y / 4 == 3) {
		*matrix_y = y % 4;
		if (x / 8 == 0) {
			*matrix_x = x;
		} else if (x / 8 == 1) {
			*matrix_x = x + 24;
		} else if (x / 8 == 2) {
			*matrix_x = x + 48;
		} else if (x / 8 == 3) {
			*matrix_x = x + 72;
		}
	}
}

**Reply by Ammarfrfr** [reply]
Did it work?
